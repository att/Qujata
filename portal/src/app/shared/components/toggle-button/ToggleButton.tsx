import styles from './ToggleButton.module.scss';
import cn from 'classnames';

export type ToggleButtonProps = {
  onClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
  isSelected: boolean;
  children: React.ReactNode;
};

export const ToggleButton: React.FC<ToggleButtonProps> = (props: ToggleButtonProps) => {
  return (
    <div className={styles.toggle_button_wrapper}>
      <button
        className={`${styles.toggle_button} ${props.isSelected ? styles.selected : ''}`}
        onClick={props.onClick}
      >
        {props.children}
      </button>
    </div>
  );
};
