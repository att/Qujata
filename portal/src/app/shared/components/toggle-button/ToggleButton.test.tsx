import { render, RenderResult } from '@testing-library/react';
import { ToggleButton, ToggleButtonProps } from './ToggleButton';

describe('ToggleButton', () => {
  test('renders ToggleButton', () => {
    const toggleButtonprops: ToggleButtonProps = {
      isSelected: true,
      onClick: jest.fn(),
      children: undefined,
    };
    const { container }: RenderResult = render(<ToggleButton {...toggleButtonprops} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});