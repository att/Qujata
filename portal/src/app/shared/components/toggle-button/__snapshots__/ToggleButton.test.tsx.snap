// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleButton renders ToggleButton 1`] = `
<div
  class="toggle_button_wrapper"
>
  <button
    class="toggle_button selected"
  />
</div>
`;
